docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,189 @@
+= VirtualMachine Templates
+
+VirtualMachine templates are openshift templates (https://docs.openshift.org/latest/dev_guide/templates.html) +
+that describe VirtualMachine specifications which produce the following template objects: +
+**One** OfflineVirtualMachine and PersistentVolumeClaim. +
+The following document describes the process of creating a Virtual machine template which will be managed by ManageIQ.
+
+== Uploading a Template
+
+If you have a YAML file that defines a VirtualMachine template, you can upload the template to an Openshift cluster using the CLI. 
+To upload a template to your current project’s template library, pass the YAML file with the following command: +
+
+----
+$ oc create -f <filename>
+----
+
+== Modifying an Uploaded Template
+You can edit a template that has already been uploaded to your project by using the following command: +
+
+----
+$ oc edit template <template>
+----
+
+Writing Templates
+You can define new VirtualMachine templates to make it easy to recreate all relevant objects (offlineVirtualMachine and PersistentVolumeClaim). The template will define the objects it creates along with some metadata to guide the creation of those objects.
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template
+metadata:
```

Comment:
I think we were also talking about a special ManageIQ label so that ManageIQ will pick up the template.

Reply:
At what point we would benefit from this specific label? With my limited understanding I think we need a way to be able to ask for VirtualMachine templates only.

Reply:
@pkliczewski If I remember it correctly, it was decided in an offline discussion, that ManageIQ will filter the templates based on such a label. You would simply be able to just ask for templates with that label attached to the REST call. You would still have to validate the template (e.g. that an OfflineVirtualMachine is present, that only supported parameters are there, ...).

Reply:
Thanks for clarification

Reply:
Thats what the labels section in the document is for,
The labels are at the bottom of the file on purpose, and in the form of:

labels:
template: “miq.github.io/kubvirt/vm-template”

“miq.github.io/kubvirt/vm-template” label applied to all objects created from this template
and without having a flag of true/flase

the syntax is mentioned here:
https://docs.openshift.org/latest/dev_guide/templates.html#writing-labels

If that dosent answer the needs, let me know

Reply:
> The labels are at the bottom of the file on purpose, and in the form of:

Ah ok. That does not help filtering the template themselves. You will need the label in the templates label section `^metadata.labels` to query templates for it.

Reply:
ok, 
do we still need to label the objects as mentioned above or this section can be removed ?

Reply:
The label must be on the metadata of the template - but that is already the case in line 34. (See below)

Reply:
"now" it is there.